se_denseslam/include/se/DenseSLAMSystem.h

```diff
@@ -93,49 +93,228 @@ class DenseSLAMSystem {
+     * reconstructed volume in voxels.
+     * \param[in] volume_dimension_ The x, y and z dimensions of the
+     * reconstructed volume in meters.
+     * \param[in] initPose The initial camera pose encoded in a 4x4 matrix.
+     * \param[in] pyramid TODO See ::Configuration.pyramid for more details.
+     * \param[in] config_ The pipeline options.
+     */
+    DenseSLAMSystem(uint2              inputSize,
+                    uint3              volume_resolution_,
+                    float3             volume_dimension_,
+                    Matrix4            initPose,
+                    std::vector<int> & pyramid,
+                    Configuration      config_);
+
+    /**
+     * Preprocess a single depth measurement frame and add it to the pipeline.
+     * This is the first stage of the pipeline.
+     *
+     * \param[in] inputDepth Pointer to the depth frame.
+     * \param[in] inputSize Size of the depth frame in pixels (width and
+     * height).
+     * \param[in] filterInput Whether to filter the frame using a bilateral
+     * filter to reduce the measurement noise.
+     * \return true (does not fail).
+     */
+    bool preprocessing(const ushort * inputDepth,
+                       const uint2    inputSize,
+                       const bool     filterInput);
+
+    /*
+     * TODO Implement this.
+     */
+    bool preprocessing(const ushort * inputDepth,
+                       const uchar3 * inputRGB,
+                       const uint2    inputSize,
+                       const bool     filterInput);
+
+    /**
+     * Update the camera pose. Create a 3D reconstruction from the current
+     * depth frame and compute a transformation using ICP. The 3D
+     * reconstruction of the current frame is matched to the 3D reconstruction
+     * obtained from all of the previous frames. This is the second stage of
+     * the pipeline.
+     *
+     * \param[in] k The intrinsic camera parameters. See
+     * ::Configuration.camera for details.
+     * \param[in] icp_threshold The ICP convergence threshold.
+     * \param[in] tracking_rate Process a frame every tracking_rate frames.
+     * \param[in] frame The index of the current frame (starts from 0).
+     * \return true if the camera pose was updated and false if it wasn't.
+     */
+    bool tracking(float4 k,
+                  float  icp_threshold,
+                  uint   tracking_rate,
+                  uint   frame);
+
+    /**
+     * Integrate the 3D reconstruction resulting from the current frame to the
+     * existing reconstruction. This is the third stage of the pipeline.
+     *
+     * \param[in] k The intrinsic camera parameters. See
+     * ::Configuration.camera for details.
+     * \param[in] integration_rate Integrate a 3D reconstruction every
+     * integration_rate frames. Should not be less than the tracking_rate used
+     * in tracking().
+     * \param[in] mu TSDF truncation bound. See ::Configuration.mu for more
```

Comment:
mu actually is misleading, we should change it. It should read something like bandwidth as it is also used in OFusion as the surface thickness parameter.